src/main/java/org/folio/inventory/dataimport/handlers/matching/preloaders/AbstractPreloader.java

```diff
@@ -0,0 +1,80 @@
+package org.folio.inventory.dataimport.handlers.matching.preloaders;
+
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonObject;
+
+import org.folio.DataImportEventPayload;
+import org.folio.MatchDetail;
+import org.folio.MatchProfile;
+import org.folio.processing.matching.loader.query.LoadQuery;
+import org.folio.processing.matching.loader.query.LoadQueryBuilder;
+import org.folio.processing.matching.reader.MatchValueReader;
+import org.folio.processing.matching.reader.MatchValueReaderFactory;
+import org.folio.processing.value.ListValue;
+import org.folio.processing.value.Value;
+import org.folio.rest.jaxrs.model.Field;
+import org.folio.rest.jaxrs.model.MatchExpression;
+
+public abstract class AbstractPreloader {
+
+    public CompletableFuture<LoadQuery> preload(LoadQuery query, DataImportEventPayload dataImportEventPayload) {
```

Comment:
Could you please refactor this method to increase readability?
Extract methods like:
* extractMatchProfile
* extractPreloadValues
* ...